source/kerberos.rst

```diff
@@ -0,0 +1,168 @@
+.. _kerberos:
+
+**********************************
+Setting up Kerberos authentication 
+**********************************
+
+This document provides configuration steps for setting up :ref:`kerberos-authentication` in |PSMDB|. 
+
+
+Assumptions
+====================
+
+The setup of the Kerberos server itself is out of scope of this document. Please refer to the `Kerberos documentation <https://web.mit.edu/kerberos/krb5-latest/doc/admin/install_kdc.html>`_ for the installation and configuration steps relevant to your operation system.
+
+We assume that you have successfully completed the following steps:
+
+* Installed and configured the Kerberos server
+* Added necessary `realms <https://web.mit.edu/kerberos/krb5-1.12/doc/admin/realm_config.html>`_
+* Added service, admin and user `principals <https://web.mit.edu/kerberos/krb5-1.5/krb5-1.5.4/doc/krb5-user/What-is-a-Kerberos-Principal_003f.html#What-is-a-Kerberos-Principal_003f>`_ 
+* Configured the ``A`` and ``PTR`` DNS records for every host running `mongod` instance to resolve the hostnames onto Kerberos realm.
+
+.. seealso:: 
+
+   MongoDB Documentation: `Kerberos Authentication <https://www.mongodb.com/docs/manual/core/kerberos/>`_
+
+Add user principals to |PSMDB|
+==================================================
+
+To get authenticated, users must exist both in the Kerberos and |PSMDB| servers with exactly matching names.
+
+After you defined the user principals in the Kerberos server, add them to the ``$external`` database in |PSMDB| and assigned required roles:
+
+.. code-block:: javascript
+
+   use $external
+   db.createUser({user: "demo@PERCONATEST.COM",roles: [{role: "read", db: "admin"}]})
+   
+Replace ``demo@PERCONATEST.COM`` with your username and Kerberos realm.
+
+
+Configure Kerberos keytab files
+=================================
+
+A keytab file stores the authentication keys for a service principal representing a ``mongod`` instance to access the Kerberos admin server. 
+
+After you have added the service principal to the Kerberos admin server, the entry for this principal is added to the ``/etc/krb5.keytab`` keytab file. 
+
+The ``mongod`` server must have access to the keytab file to authenticate. To keep the keytab file secure, restrict the access to it only for the user running the ``mongod`` process.
+
+1. Stop the `mongod` service
+   
+   .. code-block:: bash 
+
+      $ sudo systemctl stop mongod
+
+#. Copy the keytab file:
+   
+   .. code-block:: bash
+   
+      $ cp /etc/krb5.keytab /etc/krb5.keytab2
+
+#. Change the ownership to the keytab file
+   
+   .. code-block:: bash
+
+      $ sudo chown mongod:mongod /etc/krb5.keytab2
```

Comment:
PSMDB might not run on the same server with KDC, thus I'd rewrite this part as: 
1 Create keytab file for mongodb service or get a copy of the keytab file if you generate it on another host and place it under /etc/mongodb.keytab (I'd change name from default to avoid any confusions)
2 Change the ownership for the keytab file
sudo chown mongod:mongod /etc/mongodb.keytab